include/picongpu/particles/densityProfiles/FromOpenPMDImpl.hpp

```diff
@@ -0,0 +1,171 @@
+                auto series = ::openPMD::Series{
+                    ParamClass::filename,
+                    ::openPMD::Access::READ_ONLY,
+                    gc.getCommunicator().getMPIComm()};
+                ::openPMD::MeshRecordComponent dataset
+                    = series.iterations[ParamClass::iteration]
+                          .meshes[ParamClass::datasetName][::openPMD::RecordComponent::SCALAR];
+                auto const datasetExtent = dataset.getExtent();
+
+                // Offset of the local domain in file coordinates: global coordinates, no guards, no moving window
+                auto const& localDomain = Environment<simDim>::get().SubGrid().getLocalDomain();
+                bool readFromFile = true;
+                auto chunkOffset = ::openPMD::Offset(simDim, 0);
+                auto chunkExtent = ::openPMD::Extent(simDim, 0);
+                for(uint32_t d = 0; d < simDim; ++d)
+                {
+                    chunkOffset[d] = totalLocalDomainOffset[d] - ParamClass::offset[d];
+                    // Here we take care, as chunkExtent is unsigned type
+                    int32_t extent = std::min(
+                        static_cast<int32_t>(localDomain.size[d]),
+                        static_cast<int32_t>(datasetExtent[d] - chunkOffset[d]));
+                    if(extent <= 0)
+                        readFromFile = false;
+                    else
+                        chunkExtent[d] = extent;
+                }
+
+                using ValueType = FieldTmp::ValueType::type;
+                auto data = std::shared_ptr<ValueType>{nullptr};
+                if(readFromFile)
+                    data = dataset.loadChunk<ValueType>(chunkOffset, chunkExtent);
+
+                // This is MPI collective and so has to be done by all ranks
+                series.flush();
+
+                auto const* rawData = data.get();
+                auto const numElements
+                    = std::accumulate(std::begin(chunkExtent), std::end(chunkExtent), 1u, std::multiplies<uint32_t>());
+                auto hostDataBox = fieldBuffer.getHostBuffer().getDataBox().shift(guards);
+                /* This loop is a bit clunky, since we are copying one simDim-dimensional array into another,
+                 * and chunkExtent can be smaller than local domain size.
+                 * Here we rely on the loadChunk() returning data stored in x-y-z order.
+                 */
+                for(uint32_t linearIdx = 0u; linearIdx < numElements; linearIdx++)
+                {
+                    pmacc::DataSpace<simDim> idx;
+                    auto tmpLinearIdx = linearIdx;
+                    for(int32_t d = simDim - 1; d >= 0; d--)
+                    {
+                        idx[d] = tmpLinearIdx % chunkExtent[d];
+                        tmpLinearIdx /= chunkExtent[d];
+                    }
+                    hostDataBox(idx) = rawData[linearIdx];
+                }
```

Comment:
@psychocoderHPC do you know a better way to write it? Since we no longer assume that the chunk is same size as the buffer, it seems I can't make good use of D1 access

Reply:
It would be possible to use a 1D DataBox with the size of the loaded data if the loaded data is smaller than the local volume or with the size of the local volume if the OpenPMD data is larger.

Reply:
This should be what you need:

```
/* get the databox of the host buffer */
auto dataBox = fieldBuffer.getHostBuffer().getDataBox();
/* get a 1D access object to the databox */
using D1Box = DataBoxDim1Access<typename FieldTmp::DataBoxType>;
DataSpace<simDim> guards = fieldBuffer.getGridLayout().getGuard();


DataSpace<simDim> accessSpace;
for(int32_t d = 0; d < simDim; ++d)
{
    // chunkExtent <= localDomain.size is fulfilled by the data loading section
    accessSpace[d] = chunkExtent[d];
}

D1Box d1RAccess(dataBox.shift(guards), accessSpace);

/* copy from temporary buffer to fieldTmp host buffer */
for(uint32_t linearIdx = 0u; linearIdx < numElements; linearIdx++)
{
    d1RAccess[linearIdx].x() = rawData[linearIdx];
}
```

Reply:
It could be that in the line `D1Box d1RAccess(dataBox.shift(guards), accessSpace);` `ParamClass::offset` must be taken into account if the data from the file is describing a density shifted into positive direction. 

Maybe it must be `D1Box d1RAccess(dataBox.shift(guards + ParamClass::offset), accessSpace);` 
I am not sure about the definition of `ParamClass::offset` therefore I am confused if we need to use `ParamClass::offset` if it is positive or egate.

Reply:
@sbastrakov  could you address this too. IMO it would be easier to read if it is a simple loop without index transformations

Reply:
I do not think your suggestion works. The buffers are of different sizes

Reply:
It is no problem if the buffer has different extents bit I can add this place of code also after your PR.